src/RtAudioInterface.cpp

```diff
@@ -170,12 +184,30 @@ void RtAudioInterface::setup(bool verbose)
     auto dev_info_input  = rtAudioIn->getDeviceInfo(index_in);
     auto dev_info_output = rtAudioOut->getDeviceInfo(index_out);
 
-    if (static_cast<unsigned int>(getNumInputChannels()) > dev_info_input.inputChannels) {
-        setNumInputChannels(dev_info_input.inputChannels);
+    if (static_cast<unsigned int>(AudioInterface::getBaseInputChannel())
+        > dev_info_input.inputChannels) {
+        AudioInterface::setBaseInputChannel(dev_info_input.inputChannels);
+    }
+
+    if (static_cast<unsigned int>(AudioInterface::getNumInputChannels())
+        > dev_info_input.inputChannels
+              - static_cast<unsigned int>(AudioInterface::getBaseInputChannel()) + 1) {
+        mNumInChans = dev_info_input.inputChannels
+                      - static_cast<unsigned int>(AudioInterface::getBaseInputChannel())
+                      + 1;
+        AudioInterface::setNumInputChannels(mNumInChans);
+    }
+
+    if (mNumInChans == 2 && mInputMixMode == "mix-to-mono") {
+        AudioInterface::setNumInputChannels(1);
+    } else if (mNumInChans == 2) {
+        AudioInterface::setNumInputChannels(2);
```

Comment:
All those AudioInterface::function() are member functions not class function. To avoid confusion it is better to remove the class name.